cmd/docsync/main.go

```diff
@@ -0,0 +1,169 @@
+// Command docsync uploads a local directory to a specified
+// S3 bucket
+//
+// Usage:
+//
+//     docsync bucket bucketPrefix localDir
+//
+// where bucket is the name of your S3 bucket, bucketPrefix is the
+// S3 prefix to be applied to all uploaded files, and localDir is a
+// directory containing compiled docs (from docgenerate) or other
+// files to upload.
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"mime"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+func main() {
+	bucket := os.Args[1]
+	bucketPrefix := os.Args[2]
+	localDir := os.Args[3]
+
+	localKeys := mustListContents(localDir)
+
+	region := "us-east-1"
+	sess := session.Must(session.NewSession(aws.NewConfig().WithRegion(region)))
+	svc := s3.New(sess)
+
+	var remoteKeys []string
+	err := svc.ListObjectsPages(&s3.ListObjectsInput{
+		Bucket: aws.String(bucket),
+		Prefix: aws.String(bucketPrefix),
+	}, func(page *s3.ListObjectsOutput, lastPage bool) bool {
+		for _, obj := range page.Contents {
+			if !(*obj.Key == "docs/") {
```

Comment:
Should this be `bucketPrefix` instead of `"docs/"`? Also, does the parent folder actually exist as its own S3 object?